### PR TITLE
Chrome 42 / Firefox 40 / Safari 11.1 support Service-Worker-Allowed HTTP header

### PR DESCRIPTION
**Pull request by bsmth**
WPT https://wpt.fyi/results/service-workers/service-worker/Service-Worker-Allowed-header.https.html?label=master&label=stable&product=safari&product=firefox&product=chrome&aligned

- Firefox version: 40
  - https://bugzilla.mozilla.org/show_bug.cgi?id=1130101
- Edge version: 16
  - https://learn.microsoft.com/en-us/archive/microsoft-edge/legacy/developer/dev-guide/whats-new/edgehtml-16
- Chrome: Version 42
  - Reference / commits:
    - https://issues.chromium.org/issues/40395649
    - https://chromium.googlesource.com/chromium/src.git/+/d0dfdbbae1b2356c56e566d5880796af8cf246c1
    - Git Log: https://chromium.googlesource.com/chromium/src/+log/41.0.2272.0..42.0.2311.0?pretty=fuller&n=10000
- Safari version:
  - https://bugs.webkit.org/show_bug.cgi?id=180983
  - https://trac.webkit.org/changeset/226141/webkit
  - commit https://github.com/WebKit/WebKit/commit/032c12c52977840aa40bf33589f361f1dfd54b9d tagged Safari-605.1.21
    - according to [data, it's 11.1 ](https://github.com/mdn/browser-compat-data/blob/3d49b14e3611cba8c0831dcb2459049b6beaaa7e/browsers/safari.json#L127-L132)

### CODE REVIEW

File: http/headers/Service-Worker-Allowed.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+  "http": {
+    "headers": {
+      "Service-Worker-Allowed": {
+        "__compat": {
+          "description": "`Service-Worker-Allowed` response header to broaden path restrictions for a service worker's default scope.",
+          "spec_url": "https://w3c.github.io/ServiceWorker/#service-worker-allowed",
```
Comment (bsmth) [inline http/headers/Service-Worker-Allowed.json:7]: MDN page is incoming shortly if we want to pre-empt this:

```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/HTTP/Headers/Service-Worker-Allowed",
          "spec_url": "https://w3c.github.io/ServiceWorker/#service-worker-allowed",
```

Comment (caugner) [reply http/headers/Service-Worker-Allowed.json:7]: It'll be automatically added, let's not add it for now.

**Review by caugner — APPROVED** [review]
LGTM, thanks!

File: http/headers/Service-Worker-Allowed.json (hunk 1 above)
Comment (bsmth) [reply http/headers/Service-Worker-Allowed.json:7]: oh, I guess the linter removes it if it doesn't exist

**Comment by bsmth** [comment]
Thank you!

**Comment by skyclouds2001** [comment]
https://github.com/mdn/browser-compat-data/issues/19424 could be closed

**Pull request merged**